### PR TITLE
More features for the openGL visualizer

### PR DESCRIPTION
**Pull request by KonradBreitsprecher**
- **Screenshots**! Offline rendering of the screen content to a file without blocking the simulation.
Size is determined by the `window_size` parameter (8000x8000 is working). Or press the enter key during online visualization to save a snapshot to *scriptname_n.png* (autoincrementing n).
The image data is saved with matplotlib.pyplot.imsave, which supports multiple image file types.

```
visualizer = visualization.openGLLive(system, window_size = [500,500])
for i in range(100):
    system.integrator.run(1)
    visualizer.screenshot('screenshot_{}.png'.format(i))
```
- **Live particle properties**: DoubleClick on a particle to highlight it (inverse color) and display the properties (screenshot from the new sample script `visualization_charged.py`). Cycle through the particles with *arrow left/right*. Thanks @davidsean 

visualization_opengl_10

- **Performance**: Smarter openGL material setting and 'display lists' for reoccurring geometries (like spheres) gave a good performance boost. 'Ideal gas' with 10000 particles @ 20fps on a icp machine (also added a fps counter).

visualization_opengl_11

- **LB boundaries**: Shape bases LB boundaries can be drawn now.

- **LB grid and boundary nodes**: To see if the lattice assigned the correct nodes.

- **Cellsystem visualization**: Alongside coloring by node (thanks @fweik), possibility to draw the cell- and node grid via `draw_nodes` and `draw_cells`. See samples/visualization_cellsystem.py:

visualization_opengl_9

- Pause the online visualization started via run() with the space key.

- Removed rather strange dynamic COM lightning feature and disabled spotlight lightning by default.

### CODE REVIEW

File: src/python/espressomd/visualization_opengl.pyx [hunk 1]
```diff
@@ -2,19 +2,18 @@ from OpenGL.GLUT import *
 from OpenGL.GLU import *
 from OpenGL.GL import *
 from math import *
-import math
+#from PIL import Image
```
Comment (KaiSzuttor) [inline src/python/espressomd/visualization_opengl.pyx]: please remove comment

File: src/python/espressomd/visualization_opengl.pyx [hunk 2]
```diff
@@ -292,48 +306,163 @@ class openGLLive(object):
...
+        glReadBuffer(GL_COLOR_ATTACHMENT0)
+        data = glReadPixels(0, 0, self.specs['window_size'][0], self.specs['window_size'][1], GL_RGB, GL_FLOAT)
+
+        # RESHAPE THE DATA
+        data = np.flipud(data.reshape((data.shape[1],data.shape[0],3)))
+
+        # SAVE TO IMAGE
+        imsave(path, data)
+
+        print("Saved screenshot {}".format(path))
```
Comment (KaiSzuttor) [inline src/python/espressomd/visualization_opengl.pyx]: pls remove print statements

File: samples/visualization_charged.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import espressomd
```
Comment (KaiSzuttor) [inline samples/visualization_charged.py:1]: pls autopep8 this file

File: samples/visualization_interactive.py [hunk 4]
```diff
@@ -9,10 +9,32 @@
...
+
+N = 50
+for i in range(N):
+    system.part.add(pos=[0,0,0])
+
+system.thermostat.set_langevin(kT=1.0, gamma=1.0)
+
+# Callback for particle positions/velocities
+def spin():
+    system.part[:].pos = [[box_l*0.5,box_l*(i+1)/(N+2),box_l*0.5] for i in range(N)]
```
Comment (KaiSzuttor) [inline samples/visualization_interactive.py]: autopep8 pls

File: src/python/espressomd/visualization_opengl.pyx [hunk 5]
```diff
@@ -196,24 +199,35 @@ class openGLLive(object):
...
             'background_color': [0, 0, 0],
-            'update_fps': 30.0,
-            'periodic_images': [0, 0, 0],
+
+            'draw_fps': False,
             'draw_box': True,
             'draw_axis': True,
+            'draw_nodes': False,
+            'draw_cells': False,
+
```
Comment (KaiSzuttor) [inline src/python/espressomd/visualization_opengl.pyx:216]: Maybe remove blank lines and divide your specs into subspecs to maintain a logical structure

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]

File: samples/visualization_charged.py (hunk 3 above)
Comment (KonradBreitsprecher) [reply samples/visualization_charged.py:1]: done

File: src/python/espressomd/visualization_opengl.pyx (hunk 5 above)
Comment (KonradBreitsprecher) [reply src/python/espressomd/visualization_opengl.pyx:216]: I'd rather leave them in. With subspecs, the used would need to know which dict to address.

**Pull request merged**